pkg/typesystem/typesystem.go

```diff
@@ -187,8 +189,10 @@ func New(model *openfgav1.AuthorizationModel) *TypeSystem {
 	uncompiledConditions := make(map[string]*condition.EvaluableCondition, len(model.GetConditions()))
 	for name, cond := range model.GetConditions() {
 		uncompiledConditions[name] = condition.NewUncompiled(cond).
-			WithMaxEvaluationCost(config.DefaultMaxConditionEvaluationCost). // care should be taken here - decreasing can cause API compatibility problems
-			WithTrackEvaluationCost()
+			WithTrackEvaluationCost().
+			// care should be taken here - decreasing can cause API compatibility problems
+			WithMaxEvaluationCost(config.DefaultMaxConditionEvaluationCost).
+			WithInterruptCheckFrequency(defaultInterruptCheckFrequency)
```

Comment:
Should we expose this from `internal/server/config` as well like the one above? That way server specific configurations/behavior are more centralized.